operator-common/src/main/java/io/strimzi/operator/common/AbstractOperator.java

```diff
@@ -190,63 +194,82 @@ private String getLockName(String namespace, String name) {
 
             if (cr != null) {
                 Promise<Void> createOrUpdate = Promise.promise();
-
-                if (cr.getSpec() == null)   {
-                    InvalidResourceException exception = new InvalidResourceException("Spec cannot be null");
-
+                if (hasPauseReconciliationAnnotation(cr)) {
                     S status = createStatus();
-                    Condition errorCondition = new ConditionBuilder()
-                            .withLastTransitionTime(StatusUtils.iso8601Now())
-                            .withType("NotReady")
+                    Set<Condition> conditions = validate(cr);
+                    Condition pausedCondition = new ConditionBuilder()
+                            .withLastTransitionTime(new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ").format(new Date()))
+                            .withType("ReconciliationPaused")
                             .withStatus("True")
-                            .withReason(exception.getClass().getSimpleName())
-                            .withMessage(exception.getMessage())
                             .build();
-                    status.setObservedGeneration(cr.getMetadata().getGeneration());
-                    status.addCondition(errorCondition);
+                    conditions.add(pausedCondition);
+                    status.setConditions(new ArrayList<>(conditions));
 
-                    log.error("{}: {} spec cannot be null", reconciliation, cr.getMetadata().getName());
-                    updateStatus(reconciliation, status).onComplete(notUsed -> {
-                        createOrUpdate.fail(exception);
+                    updateStatus(reconciliation, status).onComplete(statusResult -> {
+                        if (statusResult.succeeded()) {
+                            createOrUpdate.complete();
+                        } else {
+                            createOrUpdate.fail(statusResult.cause());
+                        }
                     });
+                } else {
```

Comment:
if you do this with `if ... else if ...` it will probably reconcile better with #4411 which has a check which needs to be done first before checking the pause annotation.